packages/odyssey-react-mui/src/labs/SideNav/SideNav.tsx

```diff
@@ -332,64 +384,54 @@ const SideNav = ({
     [firstSideNavItemIdWithIsSelected],
   );
 
+  const sideNavItemContentProviderValue = useMemo(
+    () => ({ isCompact, depth: 1 }),
+    [isCompact],
+  );
+
   const processedSideNavItems = useMemo(
     () =>
       sideNavItems.map((item) => ({
         ...item,
-        children: item.children?.map((childProps) => (
-          <SideNavItemContent
-            key={childProps.id}
-            scrollRef={getRefIfThisIsFirstNodeWithIsSelected(childProps.id)}
-            {...childProps}
-          />
-        )),
+        children: item.children?.map((childProps) => {
+          return (
+            <SideNavItemContentContext.Provider
+              value={{ ...sideNavItemContentProviderValue, depth: 2 }}
```

Comment:
You have a `useMemo` at the top, but as soon as you create a new object with `{...values}`, it creates a new object reference again. We need to ensure we're only passing the correct version.

Is it not possible to do `depth - 1` when consuming the context in the context provider?

Meaning, in the thing that creates the context, we should also be consuming the context. If the context doesn't exist (root context), it will use the default value provided to `createContext`. Hope that helps!

Reply:
This would need a refactor to how we display the {children} inside the Accordion. Currently we use the same template component for all nav items and it does not have the context to which level they are in. When we address the workflow requirement to support multiple nested folders (4 levels) in the side-nav we can address this refactor